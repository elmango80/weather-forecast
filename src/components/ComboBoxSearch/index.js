import React, { useState, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { toComboBoxOption } from "helpers/municipalityHelper";

import { EuiComboBox, EuiFlexGrid, EuiFlexItem } from "@elastic/eui";
import { getForecast, removeForecast } from "redux/actions/forecast";

export default function ComboBoxSearch(props) {
  const [allOptions, setAllOptions] = useState([]);
  const [selectedOptions, setSelectedOptions] = useState([]);
  const dispatch = useDispatch();
  const { loading, municipalities, msgError } = useSelector(
    (state) => state.municipality
  );

  useEffect(() => {
    return () => {
      dispatch(removeForecast());
    };
  }, [dispatch]);

  useEffect(() => {
    const allOptionsStatic = toComboBoxOption(municipalities);

    setAllOptions(allOptionsStatic);
  }, [municipalities]);

  const onChange = (newSelectedOptions) => {
    setSelectedOptions(newSelectedOptions);

    if (newSelectedOptions > selectedOptions) {
      const { provinceId, municipalityId } = municipalities[
        newSelectedOptions[newSelectedOptions.length - 1].key
      ];

      dispatch(getForecast(provinceId, municipalityId));
    } else {
      const paths = newSelectedOptions.map((item) => item.key);

      dispatch(removeForecast(paths));
    }
  };

  return (
    <EuiFlexGrid columns={1} direction="column" gutterSize="s">
      <EuiFlexItem>
        <EuiComboBox
          {...props}
          fullWidth
          placeholder="Escriba el nombre del municipio"
          sortMatchesBy="startsWith"
          options={allOptions}
          isInvalid={msgError}
          selectedOptions={selectedOptions}
          onChange={onChange}
          isLoading={loading}
        />
      </EuiFlexItem>
    </EuiFlexGrid>
  );
}
