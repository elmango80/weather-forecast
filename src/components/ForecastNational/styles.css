.euiPanel.euiPanel--plain.panel-forecast-national {
    background-color: transparent;
    box-shadow: none;
    border: 0;
}