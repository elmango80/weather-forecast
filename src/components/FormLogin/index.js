import React from "react";
import { Link } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { Formik } from "formik";

import {
  loginWithEmailAndPassword,
  loginWithGithub,
  loginWithGoogle,
} from "redux/actions/auth";

import loginScheme from "schemes/loginScheme";

import {
  EuiButton,
  EuiFieldText,
  EuiFieldPassword,
  EuiForm,
  EuiFormRow,
  EuiSpacer,
  EuiHorizontalRule,
  EuiText,
} from "@elastic/eui";

const initValues = {
  email: "",
  password: "",
};

export default function FormLogin() {
  const dispatch = useDispatch();
  const { loading, msgError } = useSelector((state) => state.ui);

  return (
    <Formik
      initialValues={initValues}
      validationSchema={loginScheme}
      onSubmit={(values) => {
        const { email, password } = values;

        dispatch(loginWithEmailAndPassword(email, password));
      }}
    >
      {(props) => {
        const {
          values,
          errors,
          touched,
          handleBlur,
          handleChange,
          handleSubmit,
        } = props;

        return (
          <EuiForm
            className="form-auth"
            component="form"
            onSubmit={handleSubmit}
            invalidCallout="none"
            noValidate
          >
            <EuiFormRow
              label="Correo electrónico"
              isInvalid={errors.email && touched.email}
              error={errors.email}
            >
              <EuiFieldText
                placeholder="Correo electrónico"
                name="email"
                value={values.email}
                onChange={handleChange}
                onBlur={handleBlur}
                isInvalid={errors.email && touched.email}
                disabled={loading}
              />
            </EuiFormRow>
            <EuiFormRow
              label="Contraseña"
              isInvalid={errors.password && touched.password}
              error={errors.password}
            >
              <EuiFieldPassword
                placeholder="Contraseña"
                name="password"
                type="dual"
                value={values.password}
                onChange={handleChange}
                onBlur={handleBlur}
                isInvalid={errors.password && touched.password}
                disabled={loading}
              />
            </EuiFormRow>
            {msgError && (
              <EuiFormRow>
                <EuiText color="danger" size="s">
                  <p>{msgError}</p>
                </EuiText>
              </EuiFormRow>
            )}
            <EuiSpacer size="xl" />
            <EuiButton type="submit" fill fullWidth isLoading={loading}>
              Iniciar sesión
            </EuiButton>
            <EuiHorizontalRule margin="l" size="half" />
            <EuiButton
              color="text"
              iconType="logoGithub"
              fullWidth
              onClick={() => {
                dispatch(loginWithGithub());
              }}
            >
              Continuar con Github
            </EuiButton>
            <EuiSpacer size="m" />{" "}
            <EuiButton
              iconType="logoGoogleG"
              fullWidth
              onClick={() => {
                dispatch(loginWithGoogle());
              }}
            >
              Continuar con Google
            </EuiButton>
            <EuiHorizontalRule margin="xxl" />
            <EuiText textAlign="center">
              <p>
                {`¿No tienes una cuenta? `}
                <Link to="/auth/registrar">Regístrate</Link>
              </p>
            </EuiText>
          </EuiForm>
        );
      }}
    </Formik>
  );
}
