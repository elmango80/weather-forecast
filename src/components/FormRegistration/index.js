import React from "react";
import { Link } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { Formik } from "formik";

import { registerWithEmailAndPassword } from "redux/actions/auth";
import registerScheme from "schemes/registerScheme";

import {
  EuiButton,
  EuiFieldText,
  EuiFieldPassword,
  EuiForm,
  EuiFormRow,
  EuiSpacer,
  EuiText,
  EuiHorizontalRule,
} from "@elastic/eui";

const initValues = {
  completeName: "",
  email: "",
  password: "",
};

export default function FormRegistration() {
  const dispatch = useDispatch();
  const { loading, msgError } = useSelector((state) => state.ui);

  return (
    <Formik
      initialValues={initValues}
      validationSchema={registerScheme}
      onSubmit={(values) => {
        const { completeName, email, password } = values;

        dispatch(registerWithEmailAndPassword(completeName, email, password));
      }}
    >
      {(props) => {
        const {
          values,
          errors,
          touched,
          handleBlur,
          handleChange,
          handleSubmit,
        } = props;

        return (
          <EuiForm
            className="form-auth"
            component="form"
            onSubmit={handleSubmit}
            noValidate
          >
            <EuiFormRow
              label="Nombre completo"
              isInvalid={errors.completeName && touched.completeName}
              error={errors.completeName}
            >
              <EuiFieldText
                placeholder="Nombre completo"
                name="completeName"
                autoComplete="off"
                value={values.completeName}
                onChange={handleChange}
                onBlur={handleBlur}
                isInvalid={errors.completeName && touched.completeName}
                disabled={loading}
              />
            </EuiFormRow>
            <EuiFormRow
              label="Correo electrónico"
              isInvalid={errors.email && touched.email}
              error={errors.email}
            >
              <EuiFieldText
                placeholder="Correo electrónico"
                name="email"
                type="email"
                autoComplete="off"
                value={values.email}
                onChange={handleChange}
                onBlur={handleBlur}
                isInvalid={errors.email && touched.email}
                disabled={loading}
              />
            </EuiFormRow>
            <EuiFormRow
              label="Contraseña"
              isInvalid={errors.password && touched.password}
              error={errors.password}
              helpText={
                !errors.password &&
                "La contraseña debe contener mínimo 8 caracteres."
              }
            >
              <EuiFieldPassword
                placeholder="Contraseña"
                name="password"
                type="dual"
                autoComplete="off"
                value={values.password}
                onChange={handleChange}
                onBlur={handleBlur}
                isInvalid={errors.password && touched.password}
                disabled={loading}
              />
            </EuiFormRow>
            {msgError && (
              <EuiFormRow>
                <EuiText color="danger" size="s">
                  <p>{msgError}</p>
                </EuiText>
              </EuiFormRow>
            )}
            <EuiSpacer size="xl" />
            <EuiButton type="submit" fill fullWidth isLoading={loading}>
              Registrate
            </EuiButton>
            <EuiHorizontalRule margin="xxl" />
            <EuiText textAlign="center">
              <p>
                {`¿Ya tienes una cuenta? `}
                <Link to="/ingresar">Iniciar sesión</Link>
              </p>
            </EuiText>
          </EuiForm>
        );
      }}
    </Formik>
  );
}
