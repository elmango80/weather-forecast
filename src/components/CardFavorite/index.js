import React from "react";

import {
  EuiButton,
  EuiFlexGroup,
  EuiFlexItem,
  EuiPanel,
  EuiSpacer,
  EuiText,
} from "@elastic/eui";

import "./styles.css";
import "styles/weather-icons.css";
import { useDispatch } from "react-redux";
import { deleteFavorite } from "redux/actions/favorite";

function CardFavorite({
  municipalityId,
  municipalityName,
  stateSkyId,
  stateSkyDescription,
  currentTemperature,
  maxTemperature,
  minTemperature,
  docRef,
}) {
  const dispatch = useDispatch();

  return (
    <EuiPanel paddingSize="none" className="favorite__panel">
      <EuiFlexGroup
        direction="column"
        gutterSize="none"
        className="favorite__wrapper"
        responsive={false}
      >
        <EuiFlexItem className="favorite-panel__title" grow={false}>
          <EuiText className="favorite-panel__name">{municipalityName}</EuiText>
        </EuiFlexItem>
        <EuiFlexItem className="favorite-panel__content">
          <EuiFlexGroup
            direction="column"
            gutterSize="none"
            justifyContent="spaceBetween"
            responsive={false}
          >
            <EuiFlexItem>
              <EuiFlexGroup
                direction="column"
                gutterSize="none"
                justifyContent="spaceEvenly"
                responsive={false}
              >
                <EuiFlexItem grow={false}>
                  <EuiFlexGroup
                    className="favorite-panel-content__state-sky"
                    direction="column"
                    gutterSize="none"
                    justifyContent="spaceEvenly"
                    responsive={false}
                  >
                    <EuiFlexItem>
                      <div
                        className="card-content__state-sky"
                        data-id={stateSkyId}
                      />
                    </EuiFlexItem>
                    <EuiSpacer size="s" />
                    <EuiFlexItem className="favorite-panel__state-sky-description">
                      <EuiText className="favorite-panel__state-sky-description">
                        {stateSkyDescription}
                      </EuiText>
                    </EuiFlexItem>
                  </EuiFlexGroup>
                </EuiFlexItem>
                <EuiFlexItem grow={false}>
                  <EuiFlexGroup
                    gutterSize="s"
                    responsive={false}
                    alignItems="stretch"
                    justifyContent="center"
                  >
                    <EuiFlexItem grow={false}>
                      <EuiText className="favorite_panel_current-temperature">
                        {currentTemperature}
                      </EuiText>
                    </EuiFlexItem>
                    <EuiFlexItem grow={false}>
                      <EuiFlexGroup
                        gutterSize="none"
                        responsive={false}
                        direction="column"
                        alignItems="flexStart"
                      >
                        <EuiFlexItem>
                          <EuiText className="favorite_panel_max-temperature">
                            {maxTemperature}
                          </EuiText>
                        </EuiFlexItem>
                        <EuiFlexItem grow={false}>
                          <EuiText className="favorite_panel_min-temperature">
                            {minTemperature}
                          </EuiText>
                        </EuiFlexItem>
                      </EuiFlexGroup>
                    </EuiFlexItem>
                  </EuiFlexGroup>
                </EuiFlexItem>
              </EuiFlexGroup>
            </EuiFlexItem>
            <EuiFlexItem className="favorite-panel__button-remove" grow={false}>
              <EuiButton
                className="button--remove"
                onClick={async () => {
                  await Promise.all([
                    dispatch(deleteFavorite(docRef, municipalityId)),
                  ]);
                  console.log("Click remove!", `docRef ${docRef}`);
                }}
              >
                Eliminar
              </EuiButton>
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiPanel>
  );
}

export default React.memo(CardFavorite);
