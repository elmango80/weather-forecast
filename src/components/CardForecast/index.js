import React from "react";
import "./styles.css";
import "styles/weather-icons.css";

import {
  EuiButtonIcon,
  EuiCard,
  EuiFlexGroup,
  EuiFlexItem,
  EuiText,
  EuiTitle,
  EuiToolTip,
} from "@elastic/eui";

export default function CardForecast({ municipality, province }, ...rest) {
  return (
    <EuiCard
      {...rest}
      betaBadgeLabel="Nombre provincia"
      textAlign="left"
      title={
        <EuiFlexGroup
          gutterSize="l"
          alignItems="stretch"
          justifyContent="spaceBetween"
          responsive={false}
        >
          <EuiFlexItem>
            <EuiTitle size="m">
              <h1>{"NOMBRE DEL MUNICIPIO"}</h1>
            </EuiTitle>{" "}
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiToolTip content="AÃ±adir a mis favoritos">
              <EuiButtonIcon
                color="primary"
                onClick={() => {
                  console.log("Add to favorites click!");
                }}
                iconType="starEmpty"
                iconSize="xl"
                aria-label="Add to favorites"
              />
            </EuiToolTip>
          </EuiFlexItem>
        </EuiFlexGroup>
      }
      description={<></>}
    >
      <EuiFlexGroup
        alignItems="center"
        justifyContent="spaceBetween"
        responsive={false}
      >
        <EuiFlexItem grow={false}>
          <span className="temperature__current">8</span>
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <span className="state-sky" data-id="12" />
        </EuiFlexItem>
      </EuiFlexGroup>

      <EuiFlexGroup
        alignItems="flexEnd"
        justifyContent="spaceBetween"
        responsive={false}
      >
        <EuiFlexItem grow={false}>
          <EuiText textAlign="left">
            <h4>JUEVES, 14 ENE</h4>
            <div>Poco nuboso</div>
            <div className="temperature">
              <span className="temperature__maximun">13</span>
              <span className="temperature__minimun">5</span>
            </div>
          </EuiText>
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiText textAlign="right">
            <div>Humedad: 65%</div>
            <div>Viento: 14 km/h</div>
            <div>Lluvia: 0%</div>
          </EuiText>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiCard>
  );
}
