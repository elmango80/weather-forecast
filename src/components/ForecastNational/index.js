import React, { Fragment } from "react";
import { htmlIdGenerator } from "@elastic/eui/lib/services";

import { useFetchApi } from "hooks/useFetchApi";

import {
  EuiLoadingContent,
  EuiPanel,
  EuiSpacer,
  EuiTabbedContent,
  EuiText,
  EuiTitle,
} from "@elastic/eui";

import "./styles.css";

const tabContent = (paragraphs) =>
  paragraphs.map((paragraph, index) => (
    <Fragment key={index}>
      <EuiText data-test-subj="forecast-national">{paragraph}</EuiText>
      <EuiSpacer />
    </Fragment>
  ));

export default function ForecastNational() {
  const { data: forecast, loading, msgError } = useFetchApi();

  const tabs = [
    {
      id: htmlIdGenerator()(),
      name: "Hoy",
      content: (
        <>
          <EuiSpacer />
          {loading ? (
            <>
              <EuiLoadingContent data-test-subj="loading-content" lines={2} />
              <EuiLoadingContent data-test-subj="loading-content" lines={3} />
            </>
          ) : msgError ? (
            <EuiText color="danger" data-test-subj="forecast-national-error">
              {msgError}
            </EuiText>
          ) : (
            tabContent(forecast.today)
          )}
        </>
      ),
    },
    {
      id: htmlIdGenerator()(),
      name: "Mañana",
      content: (
        <>
          <EuiSpacer />
          {!loading && !msgError && tabContent(forecast.tomorrow)}
        </>
      ),
    },
  ];

  return (
    <EuiPanel className="panel-forecast-national">
      <EuiTitle>
        <span>Previsión España</span>
      </EuiTitle>
      <EuiSpacer size="l" />
      {!loading &&
        (msgError ? (
          <EuiText color="danger">msgError</EuiText>
        ) : (
          <EuiTabbedContent
            tabs={tabs}
            disabled={loading || msgError}
            initialSelectedTab={tabs[0]}
            autoFocus="selected"
          />
        ))}
    </EuiPanel>
  );
}
