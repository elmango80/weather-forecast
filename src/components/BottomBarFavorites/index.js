import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import { htmlIdGenerator } from "@elastic/eui/lib/services";

import {
  EuiBottomBar,
  EuiDragDropContext,
  euiDragDropMove,
  euiDragDropReorder,
  EuiDraggable,
  EuiDroppable,
  EuiFlexGroup,
  EuiFlexItem,
} from "@elastic/eui";

import CardFavorite from "components/CardFavorite";

import "./styles.css";

export default function BottomBarFavorites() {
  const [list, setList] = useState([]);
  const { favorites, loading } = useSelector((state) => state.favorite);
  const droppableId = htmlIdGenerator()();

  useEffect(() => {
    setList(Object.keys(favorites));
  }, [favorites]);

  const onDragEnd = ({ source, destination }) => {
    const lists = { [droppableId]: list };
    const actions = { [droppableId]: setList };

    if (source && destination) {
      if (source.droppableId === destination.droppableId) {
        const items = euiDragDropReorder(
          lists[destination.droppableId],
          source.index,
          destination.index
        );

        actions[destination.droppableId](items);
      } else {
        const sourceId = source.droppableId;
        const destinationId = destination.droppableId;
        const result = euiDragDropMove(
          lists[sourceId],
          lists[destinationId],
          source,
          destination
        );

        actions[sourceId](result[sourceId]);
        actions[destinationId](result[destinationId]);
      }
    }
  };

  return (
    <>
      {!loading && (
        <EuiBottomBar className="favorites__bottom-bar">
          <EuiFlexGroup justifyContent="spaceBetween" responsive={false}>
            <EuiFlexItem>
              <EuiDragDropContext onDragEnd={onDragEnd}>
                <EuiDroppable
                  className="euiFlexGroup favorites__bottom-bar-wrapper"
                  droppableId={droppableId}
                  direction="horizontal"
                  spacing="none"
                  style={{ display: "flex" }}
                >
                  {list.map((municipalityId, index) => {
                    const item = favorites[municipalityId];

                    if (!item) {
                      return <></>;
                    }

                    return (
                      <EuiDraggable
                        key={municipalityId}
                        index={index}
                        draggableId={municipalityId}
                        spacing="m"
                        disableInteractiveElementBlocking // Allows button to be drag handle
                      >
                        {(provided) => {
                          const { docRef } = favorites[item.municipalityId];

                          return <CardFavorite {...item} docRef={docRef} />;
                        }}
                      </EuiDraggable>
                    );
                  })}
                </EuiDroppable>
              </EuiDragDropContext>
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiBottomBar>
      )}
    </>
  );
}
