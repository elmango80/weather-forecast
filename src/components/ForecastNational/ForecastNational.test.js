import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";

import { useFetchApi } from "hooks/useFetchApi";

import { ForecastNational } from "components/ForecastNational";

jest.mock("hooks/useFetchApi");

describe("Test <ForecastNational />", () => {
  test("renders correctly", () => {
    useFetchApi.mockReturnValue({
      data: {},
      loading: true,
      msgError: null,
    });

    const wrapper = shallow(<ForecastNational />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show the national forecast for today and tomorrow", () => {
    const data = {
      today: [
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla vitae sodales magna. Nulla tristique mi id sem vulputate, non egestas lacus sodales.",
        "Aliquam tristique porttitor turpis sit amet tristique. Nulla sed pharetra magna. Phasellus elementum in felis non efficitur. Nulla facilisi.",
        "Integer sodales est id pulvinar maximus. Duis vel tellus imperdiet, tincidunt ipsum eget, tristique libero. In a diam ut arcu porta consectetur.",
      ],
      tomorrow: [
        "Etiam non viverra felis. Nullam aliquet ex quis lobortis luctus. Nulla eu ipsum et est pulvinar venenatis ut vel metus. Nunc ac hendrerit nisl. Vivamus hendrerit dictum nunc sed volutpat.",
        "Cras quis risus quis mi dictum elementum at condimentum dui. Nunc eu consequat velit. Phasellus id fringilla ligula.",
      ],
    };

    useFetchApi.mockReturnValue({
      data,
      loading: false,
    });

    const wrapper = shallow(<ForecastNational />);

    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.find("p").exists()).toBe(false);
    // expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
