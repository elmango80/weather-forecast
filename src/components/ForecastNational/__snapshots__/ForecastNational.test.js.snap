// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ForecastNational /> renders correctly 1`] = `
<EuiPanel
  className="panel-forecast-national"
>
  <EuiTitle>
    <span>
      Previsión España
    </span>
  </EuiTitle>
  <EuiSpacer
    size="l"
  />
</EuiPanel>
`;

exports[`Test <ForecastNational /> should show the national forecast for today and tomorrow 1`] = `
<EuiPanel
  className="panel-forecast-national"
>
  <EuiTitle>
    <span>
      Previsión España
    </span>
  </EuiTitle>
  <EuiSpacer
    size="l"
  />
  <EuiTabbedContent
    autoFocus="selected"
    initialSelectedTab={
      Object {
        "content": <React.Fragment>
          <EuiSpacer />
          <React.Fragment>
            <EuiText
              data-test-subj="forecast-national"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla vitae sodales magna. Nulla tristique mi id sem vulputate, non egestas lacus sodales.
            </EuiText>
            <EuiSpacer />
          </React.Fragment>
          <React.Fragment>
            <EuiText
              data-test-subj="forecast-national"
            >
              Aliquam tristique porttitor turpis sit amet tristique. Nulla sed pharetra magna. Phasellus elementum in felis non efficitur. Nulla facilisi.
            </EuiText>
            <EuiSpacer />
          </React.Fragment>
          <React.Fragment>
            <EuiText
              data-test-subj="forecast-national"
            >
              Integer sodales est id pulvinar maximus. Duis vel tellus imperdiet, tincidunt ipsum eget, tristique libero. In a diam ut arcu porta consectetur.
            </EuiText>
            <EuiSpacer />
          </React.Fragment>
        </React.Fragment>,
        "id": "today-forecast-tab",
        "name": "Hoy",
      }
    }
    tabs={
      Array [
        Object {
          "content": <React.Fragment>
            <EuiSpacer />
            <React.Fragment>
              <EuiText
                data-test-subj="forecast-national"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla vitae sodales magna. Nulla tristique mi id sem vulputate, non egestas lacus sodales.
              </EuiText>
              <EuiSpacer />
            </React.Fragment>
            <React.Fragment>
              <EuiText
                data-test-subj="forecast-national"
              >
                Aliquam tristique porttitor turpis sit amet tristique. Nulla sed pharetra magna. Phasellus elementum in felis non efficitur. Nulla facilisi.
              </EuiText>
              <EuiSpacer />
            </React.Fragment>
            <React.Fragment>
              <EuiText
                data-test-subj="forecast-national"
              >
                Integer sodales est id pulvinar maximus. Duis vel tellus imperdiet, tincidunt ipsum eget, tristique libero. In a diam ut arcu porta consectetur.
              </EuiText>
              <EuiSpacer />
            </React.Fragment>
          </React.Fragment>,
          "id": "today-forecast-tab",
          "name": "Hoy",
        },
        Object {
          "content": <React.Fragment>
            <EuiSpacer />
            <React.Fragment>
              <EuiText
                data-test-subj="forecast-national"
              >
                Etiam non viverra felis. Nullam aliquet ex quis lobortis luctus. Nulla eu ipsum et est pulvinar venenatis ut vel metus. Nunc ac hendrerit nisl. Vivamus hendrerit dictum nunc sed volutpat.
              </EuiText>
              <EuiSpacer />
            </React.Fragment>
            <React.Fragment>
              <EuiText
                data-test-subj="forecast-national"
              >
                Cras quis risus quis mi dictum elementum at condimentum dui. Nunc eu consequat velit. Phasellus id fringilla ligula.
              </EuiText>
              <EuiSpacer />
            </React.Fragment>
          </React.Fragment>,
          "id": "tomorrow-forecast-tab",
          "name": "Mañana",
        },
      ]
    }
  />
</EuiPanel>
`;
